ush/mesoscale/mesoscale_util.py

```diff
@@ -838,3 +839,491 @@ def get_completed_jobs(completed_jobs_file):
+                            indir, 
+                            'gdas.{VDATE}',
+                            '{VHOUR}',
+                            'atmos',
+                            'gdas.t{VHOUR}z.prepbufr'
+                        ))
+                    else:
+                        prepbufr_templates.append(os.path.join(
+                            indir, 
+                            'gdas.t{VHOUR}z.prepbufr'
+                        ))
+            for offset in offsets:
+                use_vdate = vdate + td(hours=int(offset))
+                use_vd = use_vdate.strftime('%Y%m%d')
+                use_vh = use_vdate.strftime('%H')
+                if obsname in ['both', 'metar']:
+                    if not already_preprocessed:
+                        template = os.path.join(
+                            indir, 
+                            'nam.{VDATE}',
+                            'nam.t{VHOUR}z.prepbufr.tm{OFFSET}'
+                        )
+                    else:
+                        template = os.path.join(
+                            indir, 
+                            'nam.t{VHOUR}z.prepbufr.tm{OFFSET}'
+                        )
+                    prepbufr_paths.append(fname_constructor(
+                        template, VDATE=use_vd, VHOUR=use_vh, OFFSET=offset
+                    ))
+        for template in prepbufr_templates:
+            prepbufr_paths.append(fname_constructor(
+                template, VDATE=vd, VHOUR=vh
+            ))
+    return np.concatenate((paths, np.unique(prepbufr_paths)))
+
+def preprocess_prepbufr(indir, fname, workdir, outdir, subsets):
+    if os.path.exists(os.path.join(outdir, fname)):
+        print(f"{fname} exists in {outdir} so we can skip preprocessing.")
+    else:
+        wd = os.getcwd()
+        os.chdir(workdir)
+        if os.path.isfile(os.path.join(indir, fname)):
+            run_shell_command(
+                [
+                    os.path.join(os.environ['bufr_ROOT'], 'bin', 'split_by_subset'), 
+                    os.path.join(indir, fname)
+                ]
+            )
+            if all([os.path.isfile(subset) for subset in subsets]):
+                run_shell_command(
+                    np.concatenate((
+                        ['cat'], subsets, ['>>', os.path.join(outdir, fname)]
+                    ))
+                )
+            else:
+                raise FileNotFoundError(
+                    f"The following prepbufr subsets do not exist in {workdir}: " 
+                    + ', '.join([subset for subset in subsets if not os.path.isfile(subset)])
+                    + ". Cannot concatenate subsets."
+                )
+        else:
+            print(
+                "WARNING: The following file does not exist: "
+                + f"{os.path.join(indir, fname)}."
+                + " Skipping split by subset."
+            )
+        os.chdir(wd)
+
+# Create a list of ccpa file paths
```

Comment:
@MarcelCaron-NOAA had said that it might be needed, so I added them all just in case.  I will defer to him to suggest whether it has to be here or if it can/should go.

Reply:
In an email, Marcel said about these lines:  "More complicated and possibly unnecessary, but [this](https://github.com/NOAA-EMC/EVS/pull/567/commits/6fcfacd387860bfd665e604a2e0f17b6979d7857) makes precip verification more flexible to missing obs".

Reply:
Though I guess I didn't add all the other components from this link other than this particular file.  I guess I can remove them?

Reply:
I removed the lines.  I don't think they are needed.